src/redux/books/bookSlice.js

```diff
@@ -1,50 +1,93 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import axios from 'axios';
 import { v4 as uuidv4 } from 'uuid';
 
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/IqNhFOuHupMUt4Y7YbIq';
+const booksEndPoint = '/books';
+
 const initialState = {
-  bookItems: [
-    {
-      item_id: 'item1',
-      title: 'The Great Gatsby',
-      author: 'John Smith',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item2',
-      title: 'Anna Karenina',
-      author: 'Leo Tolstoy',
-      category: 'Fiction',
-    },
-    {
-      item_id: 'item3',
-      title: 'The Selfish Gene',
-      author: 'Richard Dawkins',
-      category: 'Nonfiction',
-    },
-  ],
+  books: [],
+  isLoading: true,
+  error: undefined,
 };
 
+export const getResultItems = createAsyncThunk('result/getResultItems', async (thunkAPI) => {
+  try {
+    const getBooksUrl = `${url}${booksEndPoint}`;
+    const resp = await axios(getBooksUrl);
+    const myData = resp.data;
+
+    const myTransData = [];
+
+    Object.keys(myData).forEach((key) => {
+      myTransData.push(myData[key][0]);
+    });
+
+    return myTransData;
+  } catch (error) {
+    return thunkAPI.rejectWithValue('something went wrong...');
+  }
+});
+
+export const postNewBook = createAsyncThunk('result/postNewBook', async ({ myTitle, myAuthor, myCategory }, state, thunkAPI) => {
+  const postBooksCreateUrl = `${url}${booksEndPoint}`;
+  const payload = {
+    item_id: uuidv4(),
+    title: myTitle,
+    author: myAuthor,
+    category: myCategory,
+  };
+  console.log(payload);
+  try {
+    const resp = await axios.post(postBooksCreateUrl, payload);
+    console.log(resp);
+    if (resp.data === 'Created') {
+      return payload;
+    }
+    return thunkAPI.rejectWithValue('something went wrong...');
+  } catch (error) {
+    return thunkAPI.rejectWithValue('something went wrong...');
+  }
+});
+
```

Comment:
- [ ] You have not created action creators for the the API request which is to remove a book, this should in turn return a promise and should automatically dispatch an action type that will remove a book from books list and updates state, please kindly implement this functionality in your application 👍 
- below image is the error displayed when a user tries to delete a book because you are yet to handle it in your action creators 
![errr](https://github.com/alexansaa/react_bookstore_full/assets/89055502/1da039c2-1073-451b-b8a0-e9a9839a6092)